mypyc/irbuild/builder.py

```diff
@@ -286,19 +288,45 @@ def add_to_non_ext_dict(self, non_ext: NonExtClassInfo,
         key_unicode = self.load_str(key)
         self.call_c(dict_set_item_op, [non_ext.dict, key_unicode, val], line)
 
+    def gen_import_from(self, id: str, line: int, imported: List[str]) -> None:
+        self.imports[id] = None
+
+        globals_dict = self.load_globals_dict()
+        locals_dict = self.call_c(get_locals, [], line)
```

Comment:
Drop the whole `'PyEval_GetLocals` thing and just pass in `None` for that. `PyEval_GetLocals` won't do anything that's even a little bit like being correct in our setting (I think it will return the locals dict for whatever the enclosing interpreted function is?), and the docs for `__import__` reports " The standard implementation does not use its locals argument at all"

Reply:
This should be fixed in 8294834c6619ed0453ab6d2b3265a9377c435a1e. I also removed the primitive for getting the locals dict because we aren't using it anymore.